ivi-layermanagement-api/ilmControl/include/ilm_control.h

```diff
@@ -365,6 +365,27 @@ ilmErrorTypes ilm_displaySetRenderOrder(t_ilm_display display, t_ilm_layer *pLay
  */
 ilmErrorTypes ilm_takeScreenshot(t_ilm_uint screen, t_ilm_const_string filename);
 
+/**
+ * \brief Take a screenshot from the current displayed layer scene with callbacks
+ * The screenshot is saved as bmp file with the corresponding filename in case
+ * filename is non-NULL. In case callback_done or callback_error are non-NULL,
+ * they are called when the screenshot has been acquired and in case failure
+ * to acquire screenshot occured respectively.
+ * \ingroup ilmControl
+ * \param[in] screen Id of screen where screenshot should be taken
+ * \param[in] filename Optional location where the screenshot should be stored
+ * if the filename is empty and callbacks are available, function will not blocks
+ * \param[in] callback_done Optional callback called when screenshot is acquired
+ * \param[in] callback_error Optional callback called when screenshot acqusition failed
+ * \param[in] user_data Optional callback user data passed in by called
+ * \return ILM_SUCCESS if the method call was successful
+ * \return ILM_FAILED if the client can not call the method on the service.
+ */
+ilmErrorTypes ilm_takeScreenshotWithCallbacks(t_ilm_uint screen, t_ilm_const_string filename,
```

Comment:
to make the api more clear and avoid parallel access to the memory from Weston and from the client during the accessing the same memory in the done callback I would remove the filename from the api parameters 
then we can rename the api's like ilm_takeAsyncScreenshot